### PR TITLE
Fixes one-handed plant-people

### LINKED ISSUE #15401: Plants Sprite Refresh Error?
TL;DR : Grab plant with 1 hand missing breaks player, making them a plant without holding the plant

**Issue Description**:

Grabbing a Plant with a missing hand will stop the player's sprite from refreshing and leaving them as a plant

**What did you expect to happen**: 

Just a graphical issue leaving the player as a plant 

**What happened instead**:

I grabbed a plant and my sprite bugged out
**Steps to reproduce the problem**:

1. Loose a hand (I lost my right one)
2. Grab a plant
3. Become a plant

**When did the problem start happening**:

Jan 30*
Round ID 25638

**Extra information**:

Cured by grabbing a plant after getting the hand reattached

### LINKED ISSUE #15516: Permanently becoming a plant pot when picking up plant.
**Issue Description**:
You become a plant pot Icon instead of your character Icon after picking up a plant

**What did you expect to happen**: 
Expected to see a cowboy fox (my character) after I dropped the plant

**What happened instead**:
Became a bush in a pot permanently for the rest of the round.

**Why is this bad/What are the consequences**:
Can be used to gain a unfair advantage over antags/crew as this only affects the Icon as it allowed you to still manipulate items. such as guns/batons etc.

**Steps to reproduce the problem**:
1. Lose a hand/have only one hand.
2. attempt to pick up a pot. it should immediately drop afterwards since you need two hands to hold it.
3. become **PLANT**

**When did the problem start happening**:
Recently. on the date 12/02/2021

**Extra information**:

### PR DESCRIPTION
**Pull request by Gatchapod**
## What Does This PR Do
Fixes #15516
Fixes #15401
Picking up plants with one hand missing will no longer replace player's sprite permanently.

Feels a bit hacky, but it solves the issue.

## Why It's Good For The Game
Bug fixes always good.

## Changelog
:cl:
fix: Picking up a potted plant with one hand missing will no longer replace your sprite permanently
/:cl:

### CODE REVIEW

File: code/game/objects/structures/flora.dm [hunk 1]
```diff
@@ -16,7 +16,7 @@
 	icon_state = "pine_1"
 
 /obj/structure/flora/tree/pine/Initialize(mapload)
-	. = ..()
```
Comment (AffectedArc07) [inline code/game/objects/structures/flora.dm:19]: This actually breaks things. Undo this change.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/flora.dm (hunk 1 above)
Comment (Gatchapod) [reply code/game/objects/structures/flora.dm]: Done, probably should put more thinking into those changes.

File: code/game/objects/structures/flora.dm [hunk 2]
```diff
@@ -222,10 +222,13 @@
 		icon_state = "plant-36"
 
 /obj/item/twohanded/required/kirbyplants/equipped(mob/living/user)
-	. = ..()
-	var/image/I = image(icon = 'icons/obj/flora/plants.dmi' , icon_state = src.icon_state, loc = user)
-	I.override = 1
-	user.add_alt_appearance("sneaking_mission", I, GLOB.player_list)
+	..()
```
Comment (AffectedArc07) [inline code/game/objects/structures/flora.dm:225]: You still left this one
```suggestion
	. = ..()
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/flora.dm [hunk 3]
```diff
@@ -223,9 +223,12 @@
 
 /obj/item/twohanded/required/kirbyplants/equipped(mob/living/user)
 	. = ..()
-	var/image/I = image(icon = 'icons/obj/flora/plants.dmi' , icon_state = src.icon_state, loc = user)
-	I.override = 1
-	user.add_alt_appearance("sneaking_mission", I, GLOB.player_list)
+	//Override appearance only if the other hand is NOT empty, because you should hold offhand. If your hand is missing, get_inactive_hand() will return null.
+	//Cases where you hold one-handed item are handled in parents and this should never be called, so it shouldn't be a problem.
+	if(user.get_inactive_hand() != null)
```
[the remark this answers is not in the document]
Comment (Gatchapod) [reply code/game/objects/structures/flora.dm:228]: Seems to work fine, is more readable and should be safer than relying on parents to do their job correctly.

**Review by dearmochi — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]
